src/safe_apps/tests/test_signals_ff_hook_events.py

```diff
@@ -0,0 +1,358 @@
+        chain_id = fake.pyint()
+        provider = ProviderFactory.create()
+        SafeAppFactory.create(chain_ids=[chain_id], provider=provider)
+
+        provider.delete()
+
+        # Safe App Creation, Safe App Update, Provider update
+        assert len(responses.calls) == 2
+        assert isinstance(responses.calls[1], responses.Call)
+        assert responses.calls[
+            1
+        ].request.body == f'{{"type": "SAFE_APPS_UPDATE", "chainId": "{chain_id}"}}'.encode(
+            "utf-8"
+        )
+        assert responses.calls[1].request.url == "http://127.0.0.1/v1/hooks/events"
+        assert (
+            responses.calls[1].request.headers.get("Authorization")
+            == "Basic example-token"
+        )
+
+
+@override_settings(
+    FF_HOOK_EVENTS=True,
+    CGW_URL="http://127.0.0.1",
+    CGW_FLUSH_TOKEN="example-token",
+)
+class TagHookTestCase(TestCase):
+    @responses.activate
+    def test_on_tag_create_with_no_safe_app(self) -> None:
+        TagFactory.create()  # create
+
+        assert len(responses.calls) == 0
+
+    @responses.activate
+    def test_on_tag_create_with_safe_app(self) -> None:
+        chain_id = fake.pyint()
+        safe_app = SafeAppFactory.create(chain_ids=[chain_id])
+
+        TagFactory.create(safe_apps=(safe_app,))
+
+        # Safe App Creation, Safe App Update, M2M update, Tag create
+        assert len(responses.calls) == 4
+        assert isinstance(responses.calls[3], responses.Call)
+        assert responses.calls[
+            3
+        ].request.body == f'{{"type": "SAFE_APPS_UPDATE", "chainId": "{chain_id}"}}'.encode(
+            "utf-8"
+        )
+
+    @responses.activate
+    def test_on_tag_update_with_no_safe_app(self) -> None:
+        tag = TagFactory.create()  # create
+        tag.name = "Test Tag"
+
+        tag.save()  # update
+
+        assert len(responses.calls) == 0
+
+    @responses.activate
+    def test_on_tag_update_with_safe_app(self) -> None:
+        chain_id = fake.pyint()
+        safe_app = SafeAppFactory.create(chain_ids=[chain_id])
+        tag = TagFactory.create(safe_apps=(safe_app,))
+
+        tag.name = "test"
+        tag.save()
+
+        # Safe App Creation, Safe App Update, M2M update, Tag create, Tag update
+        assert len(responses.calls) == 5
+        assert isinstance(responses.calls[4], responses.Call)
+        assert responses.calls[
+            4
+        ].request.body == f'{{"type": "SAFE_APPS_UPDATE", "chainId": "{chain_id}"}}'.encode(
+            "utf-8"
+        )
+
+    @responses.activate
+    def test_on_tag_delete_with_no_safe_app(self) -> None:
+        tag = TagFactory.create()  # create
+
+        tag.delete()  # delete
+
+        assert len(responses.calls) == 0
+
+    @responses.activate
+    def test_on_tag_delete_with_safe_app(self) -> None:
+        chain_id = fake.pyint()
+        safe_app = SafeAppFactory.create(chain_ids=[chain_id])
+        tag = TagFactory.create(safe_apps=(safe_app,))
```

Comment:
Same as I commented before, what about having a test with more than one item (`safe_app` in this case), to verify iterations?

Reply:
Done in [e4134f3](https://github.com/safe-global/safe-config-service/pull/964/commits/e4134f37a519ab64f57f25deaa388d49cd15a1f9)